lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -224,3 +223,44 @@ def filter_recommended_courses(
             filtered_recommended_courses.append(course_data)
 
     return filtered_recommended_courses
+
+
+def get_programs_based_on_course(course_key, country_code, user):
+    """
+    Returns the filtered program recommendations based on course_run_key.
+    """
+    # Get the programs list bases on course key
+    programs = get_programs(course=course_key)
+
+    # check user is enrolled in different courses part of program
+    enrollable_programs = []
+    for program in programs:
+        course_keys = []
+        for courses in program.get('courses'):
+            course_keys.append(courses.get('key'))
+        enrollable_course_keys = _remove_user_enrolled_course_keys(user, course_keys)
+        if enrollable_course_keys:
+            enrollable_programs.append(program)
+
+    non_restricted_programs = []
+    for program in enrollable_programs:
+        # verify that program is restricted into user region
+        if not _has_country_restrictions(program, country_code):
+            non_restricted_programs.append(program)
+
+    max_price_program = max(non_restricted_programs, key=lambda x: x['price_ranges'][0]['total'])
+
+    program_upsell = {
```

Comment:
You are not sending pacing type. Please coordinate with Shafqat on how this is calculated.